pallets/stateful-storage/src/mock.rs

```diff
@@ -160,45 +170,52 @@ impl<BlockNumber> SchemaGrantValidator<BlockNumber> for SchemaGrantValidationHan
 
 pub struct SchemaHandler;
 impl SchemaProvider<u16> for SchemaHandler {
+	// For testing/benchmarking. Zero value returns None, Odd for Itemized, Even for Paginated
 	fn get_schema_by_id(schema_id: SchemaId) -> Option<SchemaResponse> {
-		// if schema_id == INVALID_SCHEMA_ID {
-		// 	return None
-		// }
-		// if schema_id == IPFS_SCHEMA_ID {
-		// 	return Some(SchemaResponse {
-		// 		schema_id,
-		// 		model: r#"schema"#.to_string().as_bytes().to_vec(),
-		// 		model_type: ModelType::Parquet,
-		// 		payload_location: PayloadLocation::IPFS,
-		// 	})
-		// }
-
-		Some(SchemaResponse {
-			schema_id,
-			model: r#"schema"#.to_string().as_bytes().to_vec(),
-			model_type: ModelType::AvroBinary,
-			payload_location: PayloadLocation::OnChain,
-		})
+		match schema_id {
```

Comment:
I would avoid mocking these out.

Reply:
Agree with enddy, ideally  that would give us a complete roundtrip similar to main runtime

Reply:
Hmmm... isn't the point of unit testing to only execute the unit under test and mock all external dependencies? (btw, this is how `pallet_messages` is doing it; I just copied the approach). Though personally I think we could benefit from test-specific implementations of `SchemaBenchmarkHelper` and `SchemaProvider` that kept track of created schemas using a BTreeMap or HashMap, so then we wouldn't have to do this hokey stuff. But I don't like the idea of taking a dependency on the Schemas pallet for unit testing.

Reply:
Right, then we have to agree all such tests should live in integration tests ? I think we can definitely investigate this as a spike, since many other pallets would benefit from it like messages pallet which depends on schema as well as msa etc.